### PR TITLE
Fixes bug 1261879 - Vagrant fixes

### PR DESCRIPTION
**Pull request by willkg**
* make memory default to 1024 but configurable via environment
  variable
* make puppet debug output configurable via environment variable
* add breakpad_rw postgres role

r?

### CODE REVIEW

File: Vagrantfile [hunk 1]
```diff
@@ -1,14 +1,21 @@
...
   config.vm.box = "puppetlabs/centos-7.0-64-puppet"
 
   config.vm.provider "virtualbox" do |v|
     v.name = "socorro-vm"
-    v.memory = 512
+    v.memory = VAGRANT_MEMORY
   end
 
-  if not is_jenkins
+  if not IS_JENKINS
```
Comment (willkg) [inline Vagrantfile]: I changed this to make it easier to pick out. I don't know if this is a proper convention, though.

File: puppet/vagrant/modules/socorro/manifests/init.pp [hunk 2]
```diff
@@ -138,6 +138,20 @@
...
+      cwd     => '/var/lib/pgsql',
+      command => 'sudo -u postgres psql template1 -c "create user breakpad_rw with encrypted password \'aPassword\' superuser"',
+      unless  => 'sudo -u postgres psql postgres -tAc "SELECT 1 FROM pg_roles WHERE rolname=\'breakpad_rw\'" | grep -q 1',
+      require => [
+        Package['postgresql94-server'],
+        Exec['postgres-initdb'],
+        File['pg_hba.conf'],
+        Service['postgresql-9.4']
+      ]
+  }
```
Comment (willkg) [inline puppet/vagrant/modules/socorro/manifests/init.pp]: This should be a verbatim copy of the previous exec with s/test/breakpad_rw/.

**Comment by willkg** [comment]
I'm using this now and I think @lonnen uses it periodically. But that's it given that it's been busted for a long time.

Given that, to review this, I'd:

1. eyeball the changes for "good-enough-ness"
2. yay/nay the "IS_JENKINS" change
3. r+ and move on with your busy day

**Comment by willkg** [comment]
Also, I think this has all the fixes I needed. I don't think I'll be making more changes after this unless we decide to make Vagrant our officially sanctioned dev environment to solve all needs.

**Comment by willkg** [comment]
@lonnen Can you glance at this?

**Comment by adngdb** [comment]
Looks good to me. However my Internet connection here is too weak for me to test this. Feel free to use my r+ to merge this, or wait on someone else to test it, up to you. :)

**Comment by willkg** [comment]
I went through and tested the various combinations before I made the PR.

Given that @lonnen and I are probably the only users right now, I'm going to merge this. If there are problems, it'll only affect us and we can fix them then.

Thank you!

**Comment by lonnen** [comment]
retroactive :+1: , tested locally but didn't get to it here in time

**Pull request merged**